pkg/httputil/proxy/proxy.go

```diff
@@ -65,23 +95,28 @@ func TransportFunc(req *http.Request) (*url.URL, error) {
 }
 
 // Without is a ProxyFunc for http.Transport that will always attempt a direct connection.
-func Without() http.RoundTripper {
+func Without(options ...Option) http.RoundTripper {
 	transport := copyDefaultTransport()
 	transport.Proxy = nil
-	return transport
+	return applyOptions(transport, options...)
 }
 
 // RoundTripper returns something very similar to http.DefaultTransport, but with the Proxy setting changed to use
 // the configuration supported by this package.
-func RoundTripper() http.RoundTripper {
-	return proxyTransport
+func RoundTripper(options ...Option) http.RoundTripper {
+	if len(options) == 0 {
+		return proxyTransport
+	}
+	transport := proxyTransport.Clone()
+	return applyOptions(transport, options...)
 }
 
 // RoundTripperWithTLSConfig returns a round tripper like RoundTripper(), but using a custom TLS config.
-func RoundTripperWithTLSConfig(tlsConf *tls.Config) http.RoundTripper {
-	trans := proxyTransport.Clone()
-	trans.TLSClientConfig = tlsConf
-	return trans
+// TODO(stehessel): refactor as proxy option.
```

Comment:
Probably don't want this TODO getting merged?

Reply:
was planning to do in follow up, but tacked it on to this pr.